test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1

```diff
@@ -7,11 +7,13 @@ Describe "Tests Get-Command with relative paths and wildcards" -Tag "CI" {
         # Create temporary EXE command files
         $file1 = Setup -f WildCardCommandA.exe -pass
         $file2 = Setup -f WildCardCommand[B].exe -pass
+        $file3 = Setup -f [.exe -pass
```

Comment:
These are probably best as Pester TestCases:

```powershell
BeforeAll {
    $testFileNames = @(
        'WildCardCommandA'
        'WildCardCommand[B]'
        '['
        '[a-mt]ook'
        '[ab-z]ook'
        '[a?b]'
        '[a*]x'
        'goose]'
    ) | ForEach-Object { "$_.exe" }

    foreach ($testFileName in $testFileNames)
    {
        Setup -f $testFileName -pass
        if ($IsLinux -or $IsMacOS)
        {
            /bin/chmod 777 $testFileName
        }
    }

    $fileTestCases = $testFileNames | ForEach-Object { @{ Filename = $_ } }
}

...

It "Should find the file <Filename>" -TestCases $fileTestCases {
    param([string]$Filename)

    $result = Get-Command -Name $Filename -Type Application
    $result | Should -Not -BeNullOrEmpty
    $result | Should -BeExactly $Filename
}
```

Reply:
I'd add "du[[ck]" too.

Reply:
@kwkam you're absolutely right that all the cases I suggested *are* valid wildcard patterns (except `goose]`).

I think they're worth adding as positive test cases if possible